File: frontend/src/lib/analytics.ts
export interface AnalyticsProperties {
  [key: string]: string;
}

export interface DeviceInfo {
  platform: string;
  os_version: string;
  architecture: string;
}

export interface UserSession {
  session_id: string;
  user_id: string;
  start_time: string;
  last_heartbeat: string;
  is_active: boolean;
}

export class Analytics {
  private static currentUserId: string | null = null;

  static async init(): Promise<void> {}

  static async disable(): Promise<void> {
    this.currentUserId = null;
  }

  static async isEnabled(): Promise<boolean> {
    return false;
  }

  static async track(_eventName: string, _properties?: AnalyticsProperties): Promise<void> {}

  static async identify(userId: string, _properties?: AnalyticsProperties): Promise<void> {
    this.currentUserId = userId;
  }

  static async startSession(userId: string): Promise<string | null> {
    this.currentUserId = userId;
    return `disabled_${Date.now()}`;
  }

  static async endSession(): Promise<void> {}

  static async trackDailyActiveUser(): Promise<void> {}

  static async trackUserFirstLaunch(): Promise<void> {}

  static async isSessionActive(): Promise<boolean> {
    return false;
  }

  static async getPersistentUserId(): Promise<string> {
    if (typeof window === 'undefined') {
      return 'user_disabled';
    }

    const existing = window.sessionStorage.getItem('voxora_user_id');
    if (existing) {
      this.currentUserId = existing;
      return existing;
    }

    const userId = `user_${Date.now()}`;
    window.sessionStorage.setItem('voxora_user_id', userId);
    this.currentUserId = userId;
    return userId;
  }

  static async checkAndTrackFirstLaunch(): Promise<void> {}

  static async checkAndTrackDailyUsage(): Promise<void> {}

  static getCurrentUserId(): string | null {
    return this.currentUserId;
  }

  static async getDeviceInfo(): Promise<DeviceInfo> {
    const platform = typeof navigator === 'undefined' ? 'unknown' : navigator.platform || 'unknown';
    return {
      platform,
      os_version: platform,
      architecture: 'unknown',
    };
  }

  static async trackSessionStarted(_sessionId: string): Promise<void> {}

  static async trackSessionEnded(_sessionId: string): Promise<void> {}

  static async cleanup(): Promise<void> {}

  static async trackAppStarted(): Promise<void> {}

  static async trackPageView(_pageName: string): Promise<void> {}

  static async trackButtonClick(_buttonName: string, _location?: string): Promise<void> {}

  static async trackFeatureUsed(_featureName: string): Promise<void> {}

  static async trackBackendConnection(_success: boolean, _errorType?: string): Promise<void> {}

  static async trackError(_errorType: string, _errorMessage?: string): Promise<void> {}

  static async trackMeetingDeleted(_meetingId: string): Promise<void> {}

  static async trackSettingsChanged(_settingType: string, _newValue: string): Promise<void> {}

  static async trackCopy(_contentType: string, _properties?: AnalyticsProperties): Promise<void> {}

  static async trackMeetingCompleted(_meetingId: string, _properties?: AnalyticsProperties): Promise<void> {}

  static async updateMeetingCount(): Promise<void> {}

  static async getMeetingsCountToday(): Promise<number> {
    return 0;
  }

  static async calculateDaysSince(_key: string): Promise<number> {
    return 0;
  }

  static async trackTranscriptionError(_errorType: string, _details?: string): Promise<void> {}

  static async trackTranscriptionSuccess(_provider: string, _duration?: number): Promise<void> {}

  static async trackSummaryGenerationStarted(_modelProvider: string, _modelName: string, _transcriptLength: number): Promise<void> {}

  static async trackSummaryGenerationCompleted(
    _modelProvider: string,
    _modelName: string,
    _success: boolean,
    _durationSeconds?: number,
    _errorMessage?: string
  ): Promise<void> {}

  static async trackSummaryRegenerated(_modelProvider: string, _modelName: string): Promise<void> {}

  static async trackModelChanged(_oldProvider: string, _oldModel: string, _newProvider: string, _newModel: string): Promise<void> {}

  static async trackCustomPromptUsed(_promptLength: number): Promise<void> {}
}

export default Analytics;

File: frontend/src-tauri/src/analytics/analytics.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use std::collections::HashMap;
use std::sync::Arc;
use tokio::sync::Mutex;
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AnalyticsConfig {
    pub api_key: String,
    pub host: Option<String>,
    pub enabled: bool,
}

impl Default for AnalyticsConfig {
    fn default() -> Self {
        Self {
            api_key: String::new(),
            host: None,
            enabled: false,
        }
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UserSession {
    pub session_id: String,
    pub user_id: String,
    pub start_time: DateTime<Utc>,
    pub is_active: bool,
}

impl UserSession {
    pub fn new(user_id: String) -> Self {
        let now = Utc::now();
        Self {
            session_id: format!("session_{}", Uuid::new_v4()),
            user_id,
            start_time: now,
            is_active: true,
        }
    }

    pub fn duration_seconds(&self) -> i64 {
        (Utc::now() - self.start_time).num_seconds()
    }
}

pub struct AnalyticsClient {
    config: AnalyticsConfig,
    user_id: Arc<Mutex<Option<String>>>,
    current_session: Arc<Mutex<Option<UserSession>>>,
}

impl AnalyticsClient {
    pub async fn new(config: AnalyticsConfig) -> Self {
        Self {
            config,
            user_id: Arc::new(Mutex::new(None)),
            current_session: Arc::new(Mutex::new(None)),
        }
    }

    pub async fn identify(
        &self,
        user_id: String,
        _properties: Option<HashMap<String, String>>,
    ) -> Result<(), String> {
        *self.user_id.lock().await = Some(user_id);
        Ok(())
    }

    pub async fn track_event(
        &self,
        _event_name: &str,
        _properties: Option<HashMap<String, String>>,
    ) -> Result<(), String> {
        Ok(())
    }

    pub async fn start_session(&self, user_id: String) -> Result<String, String> {
        let session = UserSession::new(user_id.clone());
        let session_id = session.session_id.clone();
        *self.user_id.lock().await = Some(user_id);
        *self.current_session.lock().await = Some(session);
        Ok(session_id)
    }

    pub async fn end_session(&self) -> Result<(), String> {
        self.current_session.lock().await.take();
        Ok(())
    }

    pub async fn track_daily_active_user(&self) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_user_first_launch(&self) -> Result<(), String> {
        Ok(())
    }

    pub async fn get_current_session(&self) -> Option<UserSession> {
        self.current_session.lock().await.clone()
    }

    pub async fn is_session_active(&self) -> bool {
        self.current_session.lock().await.is_some()
    }

    pub async fn track_meeting_started(
        &self,
        _meeting_id: &str,
        _meeting_title: &str,
    ) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_recording_started(&self, _meeting_id: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_recording_stopped(
        &self,
        _meeting_id: &str,
        _duration_seconds: Option<u64>,
    ) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_meeting_deleted(&self, _meeting_id: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_settings_changed(
        &self,
        _setting_type: &str,
        _new_value: &str,
    ) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_app_started(&self, _version: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_feature_used(&self, _feature_name: &str) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_summary_generation_started(
        &self,
        _model_provider: &str,
        _model_name: &str,
        _transcript_length: usize,
    ) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_summary_generation_completed(
        &self,
        _model_provider: &str,
        _model_name: &str,
        _success: bool,
        _duration_seconds: Option<u64>,
        _error_message: Option<&str>,
    ) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_summary_regenerated(
        &self,
        _model_provider: &str,
        _model_name: &str,
    ) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_model_changed(
        &self,
        _old_provider: &str,
        _old_model: &str,
        _new_provider: &str,
        _new_model: &str,
    ) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_custom_prompt_used(&self, _prompt_length: usize) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_meeting_ended(
        &self,
        _transcription_provider: &str,
        _transcription_model: &str,
        _summary_provider: &str,
        _summary_model: &str,
        _total_duration_seconds: Option<f64>,
        _active_duration_seconds: f64,
        _pause_duration_seconds: f64,
        _microphone_device_type: &str,
        _system_audio_device_type: &str,
        _chunks_processed: u64,
        _transcript_segments_count: u64,
        _had_fatal_error: bool,
    ) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_analytics_enabled(&self) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_analytics_disabled(&self) -> Result<(), String> {
        Ok(())
    }

    pub async fn track_analytics_transparency_viewed(&self) -> Result<(), String> {
        Ok(())
    }

    pub fn is_enabled(&self) -> bool {
        let _ = &self.config;
        false
    }

    pub async fn set_user_properties(
        &self,
        _properties: HashMap<String, String>,
    ) -> Result<(), String> {
        Ok(())
    }
}

pub async fn create_analytics_client(config: AnalyticsConfig) -> AnalyticsClient {
    AnalyticsClient::new(config).await
}
